src/fsharp/SyntaxTreeOps.fs

```diff
@@ -566,9 +569,10 @@ let mkSynBindingRhs staticOptimizations rhsExpr mRhs retInfo =
         | None -> rhsExpr, None
     rhsExpr, retTyOpt
 
-let mkSynBinding (xmlDoc, headPat) (vis, isInline, isMutable, mBind, spBind, retInfo, mEquals, origRhsExpr, mRhs, staticOptimizations, attrs, memberFlagsOpt) =
+let mkSynBinding (xmlDoc: PreXmlDoc, headPat) (vis, isInline, isMutable, mBind, spBind, retInfo, mEquals, origRhsExpr, mRhs, staticOptimizations, attrs, memberFlagsOpt) =
     let info = SynInfo.InferSynValData (memberFlagsOpt, Some headPat, retInfo, origRhsExpr)
     let rhsExpr, retTyOpt = mkSynBindingRhs staticOptimizations origRhsExpr mRhs retInfo
+    let mBind = unionRangeWithXmlDoc xmlDoc mBind
```

Comment:
Could you explain more - why should the range of the XML docs be included in the range for the binding?

Reply:
We want to be able to analyze comment blocks for declarations/bindings in a unified manner. It is most convenient not to calculate the beginning of the declaration/binding depending on the presence/absence of XmlDoc, but to know that XmlDoc is part of the binding, that is the range of XmlDoc is included in the binding range.

Reply:
I think it's likely that other parts of the compilation logic and tests (e.g. error messages, maybe debug points, editor tooling) depend on the range not including the comment

Reply:
At the moment, attributes [are included](https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWQDaAPAHKsALnwCuNGgD4AugB0ATjOQ1YgsAA8wAXjABmSCAC%2BQA) in the top binding range. Could this be an argument for including XmlDoc in range as well? (:

And do you think there is a neat way to add XmlDoc to the range for local bindings?